src/dmtcpplugin.cpp

```diff
@@ -187,8 +185,8 @@ dmtcp_set_ckpt_dir(const char *dir)
 EXTERNC const char *
 dmtcp_get_coord_ckpt_dir(void)
 {
-  static string dir = CoordinatorAPI::getCoordCkptDir();
-  return dir.c_str();
+  static string *dir = new string(CoordinatorAPI::getCoordCkptDir());
```

Comment:
Doesn't this create a memory leak by allocating in the heap, instead of tmpdir on the stack?